### ISSUE TITLE
Need a binding redirect in MSBuild.exe.config for System.IO.Compression.dll to 4.1.2.0

### ISSUE THREAD

**Issue by KirillOsenkov** [Microsoft/msbuild#2234]
We're seeing inline tasks (compiled with Code Task Factory) that use System.IO.Compression crash with weird exceptions because both 4.0.0.0 and 4.1.2.0 versions of System.IO.Compression.dll are loaded into the same MSBuild process.

We added 4.1.2.0 for the binary logger, but for some reason the code task factories will pick up the 4.0.0.0 from the GAC. Example:
https://github.com/mono/monodevelop/blob/master/main/msbuild/DownloadFile.targets#L10

**Comment by rainersigwald** [comment]
What's the exception? I wouldn't expect types to collide from our internal use vs task internals.

**Comment by KirillOsenkov** [comment]
```
C:\monodevelop\main\msbuild\DownloadFile.targets(16,5): error MSB4018: The "MDDownloadFiles" task failed unexpectedly.
This is an unhandled exception from a task -- PLEASE OPEN A BUG AGAINST THE TASK OWNER.
System.ArrayTypeMismatchException: Attempted to access an element as a type incompatible with the array.
   at System.Collections.Generic.List`1.Add(T item)
   at System.IO.Compression.ZipArchive.AddEntry(ZipArchiveEntry entry)
   at System.IO.Compression.ZipArchive.ReadCentralDirectory()
   at System.IO.Compression.ZipArchive.get_Entries()
   at MDBuildTasks.MDDownloadFiles.UnpackIfChanged(String srcFile, String destDir) in c:\Users\kirillo\AppData\Local\Temp\1oqejkwk.0.cs:line 126
   at MDBuildTasks.MDDownloadFiles.DownloadFile(String cacheDir, String url, String sha1, String outputDir, String outputName, Boolean unpack) in c:\Users\kirillo\AppData\Local\Temp\1oqejkwk.0.cs:line 85
   at MDBuildTasks.MDDownloadFiles.Execute() in c:\Users\kirillo\AppData\Local\Temp\1oqejkwk.0.cs:line 51
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() in C:\MSBuild\src\Build\BackEnd\TaskExecutionHost\TaskExecutionHost.cs:line 631
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext() in C:\MSBuild\src\Build\BackEnd\Components\RequestBuilder\TaskBuilder.cs:line 787 [C:\monodevelop\main\src\addins\MonoDevelop.AzureFunctions\MonoDevelop.AzureFunctions.csproj]
```

**Comment by rainersigwald** [comment]
Fixed by #2235.
